dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/KeyJsonValueController.java

```diff
@@ -177,21 +135,14 @@ public void deleteNamespace( @PathVariable String namespace, HttpServletResponse
         HttpServletResponse response )
         throws Exception
     {
-        KeyJsonValue keyJsonValue = validateAndGetValue( namespace, key );
-
-        if ( !aclService.canRead( currentUserService.getCurrentUser(), keyJsonValue ) )
-        {
-            throw new WebMessageException(
-                WebMessageUtils.forbidden(
-                    "You do not have the authority to access the key: '" + key + "' in the namespace:'" + namespace +
-                        "'" ) );
-        }
-
-        KeyJsonValue metaDataValue = new KeyJsonValue();
-        BeanUtils.copyProperties( metaDataValue, keyJsonValue );
-        metaDataValue.setValue( null );
```

Comment:
This would actually not prevent the value from being send because while this clears the `value` field `getValue()` does return the `jbPlainValue` or `encryptedValue` field in case the `value` field is `null`.
So this bug has always existed. You can check this for example here https://debug.dhis2.org/dev/api/dataStore/ANDROID_SETTING_APP/program_settings/metaData (might need login first)